include/trial/protocol/json/partial/skip.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef TRIAL_PROTOCOL_JSON_PARTIAL_SKIP_HPP
+#define TRIAL_PROTOCOL_JSON_PARTIAL_SKIP_HPP
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (C) 2018 Vinícius dos Santos Oliveira <vini.ipsmaker@gmail.com>
+//
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <trial/protocol/json/reader.hpp>
+
+namespace trial
+{
+namespace protocol
+{
+namespace json
+{
+namespace partial
+{
+
+template<class CharT>
+typename basic_reader<CharT>::view_type
+skip(basic_reader<CharT> &reader, std::error_code &ec)
+{
+    using view_type = typename basic_reader<CharT>::view_type;
+    using size_type = typename view_type::size_type;
+
+    switch (reader.symbol()) {
+    case token::symbol::end:
+    case token::symbol::error:
```

Comment:
For `error` I would assign `reader::error()` to `ec` to get the underlying cause.

Reply:
But then the user will never be able to distinguish between successful skip that had an error on the next token and the rest. If he wants to check the underlying error of `insufficient_tokens`, he can just call `reader.error()` himself. No information loss.

Reply:
Good point